fragment/src/main/java/com/pascalwelsch/compositeandroid/fragment/FragmentDelegate.java

```diff
@@ -41,1144 +45,1560 @@ public FragmentDelegate(final ICompositeFragment icompositefragment) {
 
-            }
-        }, new SuperCall<Context>() {
             @Override
-            public Context call(final Object... args) {
-                return getOriginal().super_getContext();
+            public Context call() {
+                if (iterator.hasPrevious()) {
+                    return iterator.previous().getContext(this);
+                } else {
+                    return getOriginal().super_getContext();
+                }
             }
-        });
+        };
+        return superCall.call();
     }
 
-
     public Object getEnterTransition() {
-        return callFunction("getEnterTransition()", new PluginCall<FragmentPlugin, Object>() {
-            @Override
-            public Object call(final NamedSuperCall<Object> superCall, final FragmentPlugin plugin,
-                    final Object... args) {
+        if (mPlugins.isEmpty()) {
+            return getOriginal().super_getEnterTransition();
+        }
 
-                return plugin.getEnterTransition(superCall);
+        final ListIterator<FragmentPlugin> iterator = mPlugins.listIterator(mPlugins.size());
+
+        final CallFun0<Object> superCall = new CallFun0<Object>("getEnterTransition()") {
 
-            }
-        }, new SuperCall<Object>() {
             @Override
-            public Object call(final Object... args) {
-                return getOriginal().super_getEnterTransition();
+            public Object call() {
+                if (iterator.hasPrevious()) {
+                    return iterator.previous().getEnterTransition(this);
+                } else {
+                    return getOriginal().super_getEnterTransition();
+                }
             }
-        });
+        };
+        return superCall.call();
     }
 
-
     public Object getExitTransition() {
-        return callFunction("getExitTransition()", new PluginCall<FragmentPlugin, Object>() {
-            @Override
-            public Object call(final NamedSuperCall<Object> superCall, final FragmentPlugin plugin,
-                    final Object... args) {
+        if (mPlugins.isEmpty()) {
+            return getOriginal().super_getExitTransition();
+        }
 
-                return plugin.getExitTransition(superCall);
+        final ListIterator<FragmentPlugin> iterator = mPlugins.listIterator(mPlugins.size());
+
+        final CallFun0<Object> superCall = new CallFun0<Object>("getExitTransition()") {
 
-            }
-        }, new SuperCall<Object>() {
             @Override
-            public Object call(final Object... args) {
-                return getOriginal().super_getExitTransition();
+            public Object call() {
+                if (iterator.hasPrevious()) {
+                    return iterator.previous().getExitTransition(this);
+                } else {
+                    return getOriginal().super_getExitTransition();
+                }
             }
-        });
+        };
+        return superCall.call();
```

Comment:
I'm not sure I understand why you are allocating a function just to call it immediately. Is there something I'm missing?